trappy/base.py

```diff
@@ -265,6 +265,24 @@ def generate_parsed_data(self):
 
             yield data_dict
 
+    def optimize_dataframe(self):
+        """Optimize memory footprint by setting minimal data types required by
+           each column"""
+        for col in self.data_frame.columns:
+            if self.data_frame[col].dtype.kind == 'i':
+                self.data_frame[col].apply(pd.to_numeric, downcast='signed')
+                continue
+            if self.data_frame[col].dtype.kind == 'f':
+                self.data_frame[col].apply(pd.to_numeric, downcast='float')
+                continue
+            # Convert string objects (pointer) to categories, only when we have
```

Comment:
Shouldn't we do this?
```
if self.data_frame[col].dtype.kind == 'S':
    # string optimization here
else:
    continue

Reply:
Well, for trace events we have either integers, floats or strings... do we have other values to consider?

Reply:
There are arrays... but I'm not sure how they are represented and parsed by trappy.
@JaviMerino should know it better...

Reply:
I seem to remember some thermal event parsing being somewhat botched turning cpumask into a weird type (or maybe it might just have turned them into weird looking strings).

Reply:
arrays in a trace are expanded. `load=[0, 20, 42, 50]` becomes `load0=0 load1=20 load2=42 load3=50`. So the dataframe will see them as columns with numbers and you optimizations apply.